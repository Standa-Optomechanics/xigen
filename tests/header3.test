#!/bin/sh
. $srcdir/defs

cat < $file_expected > $file_ok

expect_out_header_wdocs
expect_consistent
