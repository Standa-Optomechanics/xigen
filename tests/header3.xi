protocol "v1"
defaults with crc, answer, public

/**
	* \english
	* @name State
	* Describes motor state
	* \endenglish
	* \russian
	* @name Состояние
	* Состояние мотора
	* \endrussian
	*/
flagset DevStateFoobar:
	DEV_STATE_FOOBAR_OFF 		= 0x00	/**< \english motor is turned off. \endenglish \russian мотор отключен от контроллера. \endrussian */
	DEV_STATE_FOOBAR_STOP 		= 0x01	/**< \english motor is stopped. \endenglish \russian мотор остановлен. \endrussian */
	DEV_STATE_FOOBAR_MOVING 	= 0x02	/**< \english motor is moving. \endenglish \russian мотор работает. \endrussian */

command "foobar" reader "gfbr" (13)
with inline
fields:
  int8u flag flags 
  reserved 6

command "baz" reader "gfbr" (13)
with inline
fields:
  int8u flag flags
  reserved 6

/** vim: set ft=c: */
