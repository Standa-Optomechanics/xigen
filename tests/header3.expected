
/** @cond DO_NOT_WANT */
#define LIBXIMC_PROTOCOL_VERSION 1
/** @endcond */


/*
 ------------------------------------------
   BEGIN OF GENERATED struct declarations
 ------------------------------------------
*/

/**
	* @anchor flagset_devstatefoobar
	* \english
	* @name State
	* Describes motor state
	* \endenglish
	* \russian
	* @name Состояние
	* Состояние мотора
	* \endrussian
	*/
//@{
#define DEV_STATE_FOOBAR_OFF      0x00 	/**< \english motor is turned off. \endenglish \russian мотор отключен от контроллера. \endrussian */
#define DEV_STATE_FOOBAR_STOP     0x01 	/**< \english motor is stopped. \endenglish \russian мотор остановлен. \endrussian */
#define DEV_STATE_FOOBAR_MOVING   0x02 	/**< \english motor is moving. \endenglish \russian мотор работает. \endrussian */
//@}


/*
 --------------------------------------------
   BEGIN OF GENERATED function declarations
 --------------------------------------------
*/
	result_t XIMC_API get_foobar (device_t id, unsigned int* flags);

	result_t XIMC_API get_baz (device_t id, unsigned int* flags);


/*
 -------------------------
   END OF GENERATED CODE
 -------------------------
*/
